libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/camera/NavigationCamera.java

```diff
@@ -47,6 +49,7 @@
 public class NavigationCamera implements LifecycleObserver {
 
   private static final int ONE_POINT = 1;
+  private final Set<TrackingModeTransitionListener> trackingModeTransitionListeners = new HashSet<>();
```

Comment:
I'd suggest using a `CopyOnWriteArrayList` over here so that a user can call `removeTrackingModeTransitionListener` from inside of the `TrackingModeTransitionListener` callback without running into `ConcurrentModificationException`.